app/main.py

```diff
@@ -1,3 +1,26 @@
 def cache(func):
-    # Write your code here
-    pass
+    caches = []
+
+    def inner(*args):
+        local_cache = []
+        for arg in args:
+            local_cache.append(arg)
+        if local_cache in caches:
+            print("Getting from cache")
+        else:
+            print("Calculating new result")
+            caches.append(local_cache)
+
+        return func(*args)
+
+    return inner
+
+
+@cache
+def long_time_func(a, b, c):
+    return (a ** b ** c) % (a * c)
+
+
+@cache
+def long_time_func_2(n_tuple, power):
+    return [number ** power for number in n_tuple]
```

Comment:
there is no need to write functions and decorate them. if it is for testing, still better not to push this